product_customer_code/models/product.py

```diff
@@ -0,0 +1,31 @@
+# coding: utf-8
+# Copyright 2017 Vauxoo (https://www.vauxoo.com) info@vauxoo.com
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from openerp import api, fields, models
+
+
+class ProductProduct(models.Model):
+
+    _inherit = "product.product"
+
+    product_customer_code_ids = fields.One2many(
+        'product.customer.code', 'product_id', string='Product customer codes',
+        copy=False, help="Different codes that has the product for each "
+        "partner.")
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=80):
+        res = super(ProductProduct, self).name_search(
+            name, args=args, operator=operator, limit=limit)
+        if not res:
```

Comment:
Change in [3600172](https://github.com/Vauxoo/addons-vauxoo/pull/1206/commits/36001728dab8f0e9c9c524a7fd4de0eec1b2f010)